packages/node_modules/@ciscospark/media-engine-webrtc/src/engine.js

```diff
@@ -0,0 +1,691 @@
+import {debounce} from 'lodash-decorators';
+import {nonenumerable} from 'core-decorators';
+import {make as makeTemplateContainer, oneFlight, tap, whileInFlight} from '@ciscospark/common';
+import {evented} from '@ciscospark/common';
+import Events from 'ampersand-events';
+import {defaults} from 'lodash';
+import {parse} from 'sdp-transform';
+import {
+  ensureH264,
+  limitBandwith,
+  getMediaDirectionFromTracks,
+  kindToPropertyFragment
+} from './webrtc-helpers';
+
+const DirectionContainer = makeTemplateContainer(WeakMap, Map);
+const targetMediaDirection = new DirectionContainer();
+
+const capitalize = {
+  audio: `Audio`,
+  video: `Video`
+};
+
+/**
+ * Wrapper around targetMediaDirection.get which return `inactive` instead of
+ * undefined
+ * @param {WebRTCMediaEngine} target
+ * @param {string} kind
+ * @private
+ * @returns {string}
+ */
+function getTargetMediaDirection(target, kind) {
+  return targetMediaDirection.get(target, kind) || `inactive`;
+}
+
+/**
+ * Interface for doing webrtc things
+ * @protected
+ */
+export default class WebRTCMediaEngine {
+  /**
+   * Wrapper around navigator.mediaDevices.getUserMedia
+   *
+   * @param {MediaStreamContraints} constraints
+   * @returns {Promise<MediaStream>}
+   */
+  static getUserMedia(constraints) {
+    const finalConstraints = defaults({}, constraints, {fake: process.env.NODE_ENV === `test`});
```

Comment:
How does this work with Chrome's fake media flags? Do we need to set both the flags and the NODE_ENV?

Reply:
this only works in firefox; in chrome, you need to launch with the appropriate switches.